modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceUnitsOfMeasure.testcase

```diff
@@ -440,6 +440,80 @@ definition {
 		}
 	}
 
+	@description = "COMMERCE-12393. As a buyer I can view the UOMs sorted by priority on the UOM selector from the product details page."
+	@priority = 4
+	test CanAssertUOMsAreSortedByPriority {
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("And an account with a buyer user is created") {
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And 4 UOMs with decreasing priority are created and assigned to a SKU (e.g. MIN93015)") {
+			for (var indexAndActivation : list "4:true,1:true,2:false,3:true") {
+				var index = StringUtil.extractFirst(${indexAndActivation}, ":");
+				var active = StringUtil.extractLast(${indexAndActivation}, ":");
+
+				CommerceJSONSKUsAPI._addCommerceUOM(
+					active = ${active},
+					basePrice = 25,
+					priority = ${index},
+					productSKU = "MIN93015",
+					uomKey = "uom${index}",
+					uomName = "UOM${index}");
+			}
+		}
+
+		task ("When the buyer user logs in ") {
+			User.logoutPG();
+
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+		}
+
+		task ("And the buyer goes to the product details page") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/p/abs-sensor");
+		}
+
+		task ("Then the buyer user can assert the UOMs are sorted by priority ") {
+			for (var indexAndUOMName : list "1:uom1,2:uom3,3:uom4") {
+				var index = StringUtil.extractFirst(${indexAndUOMName}, ":");
+				var uomName = StringUtil.extractLast(${indexAndUOMName}, ":");
+
+				AssertElementPresent(
+					index = ${index},
+					locator1 = "CommerceFrontStore#FS_PRODUCT_UOM_SELECTOR_VALUE_BY_INDEX",
+					uomName = ${uomName});
```

Comment:
- The highest priority UOM should be pre-selected(UOM2). I noticed that we are just checking with an AssertElementPresent and an Idex, but we don't verify the content of what is written. At that moment, you should be able to use the AssertSelectedLabel with this path here: CommerceFrontStore#FS_PRODUCT_UOM_SELECT